### ISSUE TITLE
[bug] Forked workload not ready for Rollouts

### ISSUE THREAD

**Issue by mikhail-klimko** [signadot/community#50]
**Describe the bug**
A simple `Sandbox` with forked Rollouts stuck in `ForkedWorkloadNotReady` state

**To Reproduce**
Signadot operator version: `signadot/operator:v0.16.0`
Argo Rollout version: `argo-rollouts:v1.6.1`
Kubernetes version: EKS (v1.27 k8s)

1. Apply a test rollout with service (and virtual service if you have istio):

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Rollout
metadata:
  name: canary-demo
spec:
  replicas: 5
  revisionHistoryLimit: 1
  selector:
    matchLabels:
      app: canary-demo
  template:
    metadata:
      labels:
        app: canary-demo
    spec:
      containers:
        - name: canary-demo
          image: docker.io/mklimko/rollouts-demo:blue
          imagePullPolicy: Always
          ports:
            - name: http
              containerPort: 8080
              protocol: TCP
          resources:
            requests:
              memory: 32Mi
              cpu: 5m
  strategy:
    canary:
      steps:
        - setWeight: 20
        - pause: {}
---
apiVersion: v1
kind: Service
metadata:
  name: canary-demo
spec:
  ports:
  - port: 80
    targetPort: http
    protocol: TCP
    name: http
  selector:
    app: canary-demo
---
apiVersion: networking.istio.io/v1alpha3
kind: VirtualService
metadata:
  name: canary-demo
spec:
  hosts:
  - canary-demo.signadot.svc.cluster.local
  http:
  - route:
    - destination:
        host: canary-demo
        port:
          number: 80
```

2. Make sure the rollout is healthy:

```yaml
status:
  HPAReplicas: 5
  availableReplicas: 5
  blueGreen: {}
  canary: {}
  conditions:
  - lastTransitionTime: "2024-03-11T11:57:48Z"
    lastUpdateTime: "2024-03-11T11:57:48Z"
    message: RolloutCompleted
    reason: RolloutCompleted
    status: "True"
    type: Completed
  - lastTransitionTime: "2024-03-11T12:17:10Z"
    lastUpdateTime: "2024-03-11T12:17:10Z"
    message: Rollout is healthy
    reason: RolloutHealthy
    status: "True"
    type: Healthy
  - lastTransitionTime: "2024-03-11T11:57:48Z"
    lastUpdateTime: "2024-03-11T12:17:10Z"
    message: ReplicaSet "canary-demo-f794cd79f" has successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  - lastTransitionTime: "2024-03-11T12:17:10Z"
    lastUpdateTime: "2024-03-11T12:17:10Z"
    message: Rollout has minimum availability
    reason: AvailableReason
    status: "True"
    type: Available
```

3. Create a simple sandbox

```yaml
name: sbx-rollout-test
spec:
  cluster: non-prod-ue1-test-1
  forks:
  - forkOf:
      kind: Rollout
      namespace: signadot
      name: canary-demo
    customizations:
      images:
        - container: canary-demo
          image: docker.io/mklimko/rollouts-demo:green
  defaultRouteGroup:
    endpoints:
    - name: canary-demo
      target: http://canary-demo.signadot.svc.cluster.local:80
```

**Expected behavior**

Expecting the sandbox to be marked as Ready/Healthy.
However, see observed behavior next

**Observed behavior**

Sandbox marked as Unhealthy

`SignadotSandbox` spec's status:
```yaml
status:
  conditions:
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    message: 'ForkedWorkload sbx-rollout-test-ro-canary-demo is not Ready: ForkedWorkloadNotReady:
      Forked workload not ready.'
    reason: ForkedWorkloadNotReady
    status: "False"
    type: Ready
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    message: Sandbox has not been processed
    reason: Default
    status: "False"
    type: ResourcesProvisioned
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    message: All desired resources have been created.
    reason: ResourceCreated
    status: "True"
    type: ResourcesCreated
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    message: All customizations applied successfully
    reason: Reconciled
    status: "True"
    type: CustomizationApplied
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    message: All desired external workloads have connected.
    reason: ExternalWorkloadConnected
    status: "True"
    type: ExternalWorkloadReady
  - lastTransitionTime: "2024-03-11T12:02:50Z"
    message: Routing is ready
    reason: RoutingReady
    status: "True"
    type: RoutingConfigReady
```

signadot cli output:
```console
signadot sandbox apply -f rollout-fork-test.yaml
Created sandbox "sbx-rollout-test" (routing key: <key>) in cluster "non-prod-ue1-test-1".

Waiting (up to --wait-timeout=3m0s) for sandbox to be ready...
✗ Sandbox status: Not Ready: ForkedWorkload sbx-rollout-test-ro-canary-demo is not Ready: ForkedWorkloadNotReady: Forked workload not ready.

Dashboard page: https://app.signadot.com/sandbox/id/<id>

SANDBOX ENDPOINT   TYPE   URL
canary-demo        host   https://canary-demo--sbx-rollout-test.preview.signadot.com

The sandbox was applied, but it may not be ready yet. To check status, run:

  signadot sandbox get sbx-rollout-test

Error: timed out after 3m0s
```

`ForkedWorkload` spec's status:
```yaml
status:
  baseline:
    generation: 1
    workloadID: 891736bd-2d47-4d00-b0fe-42a8b8926ab5
  conditions:
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    message: Forked workload not ready.
    reason: ForkedWorkloadNotReady
    status: "False"
    type: Ready
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    message: All customizations applied successfully
    reason: Reconciled
    status: "True"
    type: CustomizationApplied
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    message: All services created successfully
    reason: Reconciled
    status: "True"
    type: ServicesCreated
  fork:
    objectRef:
      apiVersion: argoproj.io/v1alpha1
      kind: Rollout
      name: sbx-rollout-test-ro-canary-demo-2d755bee
      namespace: signadot
  observedGeneration: 1
  routingConfig:
    devMeshSidecarPresent: false
    services:
    - from: canary-demo
      ports:
      - port: 80
        targetPort: 8080
      to: sbx-rollout-test-ro-canary-demo-242c125a
```

**Additional context**

**Forked** rollout is healthy!
```yaml
status:
  HPAReplicas: 1
  availableReplicas: 1
  blueGreen: {}
  canary: {}
  conditions:
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    lastUpdateTime: "2024-03-11T12:02:49Z"
    message: RolloutCompleted
    reason: RolloutCompleted
    status: "True"
    type: Completed
  - lastTransitionTime: "2024-03-11T12:16:02Z"
    lastUpdateTime: "2024-03-11T12:16:02Z"
    message: Rollout is healthy
    reason: RolloutHealthy
    status: "True"
    type: Healthy
  - lastTransitionTime: "2024-03-11T12:02:49Z"
    lastUpdateTime: "2024-03-11T12:16:02Z"
    message: ReplicaSet "sbx-rollout-test-ro-canary-demo-2d755bee-84bbb4dbc9" has
      successfully progressed.
    reason: NewReplicaSetAvailable
    status: "True"
    type: Progressing
  - lastTransitionTime: "2024-03-11T12:16:02Z"
    lastUpdateTime: "2024-03-11T12:16:02Z"
    message: Rollout has minimum availability
    reason: AvailableReason
    status: "True"
    type: Available
```

And preview endpoint is accessible too!
<img width="498" alt="image" src="https://github.com/signadot/community/assets/67968402/56072560-00a3-4452-ad60-0187f60b5be6">

**Comment by foxish** [comment]
@mikhail-klimko - thanks for the report, we'll take a look at this and post our findings once we have a repro on our end.

**Comment by daniel-de-vera** [comment]
@mikhail-klimko, we have identified and fixed the bug.
It will be included in the next operator release.

**Comment by foxish** [comment]
@mikhail-klimko - This fix has been incorporated into v0.16.1 that was just released: https://www.signadot.com/docs/release-notes#fixed. Please reopen this issue if it persists after you upgrade.
